train.py

```diff
@@ -19,6 +19,7 @@
 
 import torch
 import torch.nn.functional as F
+from float8_experimental.fsdp_utils import precompute_float8_dynamic_scale_for_fsdp
```

Comment:
@weifengpy I think we should hide this to the path where `enable_fp8_allgather` path happened? 

The problem here is that for every feature that requires an additional install from other dependency, we should try to hide the import to the path that uses it instead of import it globally, otherwise for users who didn't install the float8_experimental, if they rebase, and it would just fail to train for them.

Please submit a follow up PR to fix this

Reply:
got you. I am moving it from top-level to if-else now  https://github.com/pytorch/torchtitan/pull/464

thanks for the timely reminder